src/PluginClient.php

```diff
@@ -62,6 +62,36 @@ public function __construct($client, array $plugins = [], array $options = [])
     }
 
     /**
+     * Adds a plugin to the client.
+     *
+     * @param Plugin $plugin
+     */
+    public function addPlugin(Plugin $plugin)
```

Comment:
How about renaming to "appendPlugin". Because you are **appending** items to the **end** of a queue. It would be more clear that the order of the plugins is important.